x-pack/legacy/plugins/infra/public/pages/infrastructure/metrics_explorer/index.tsx

```diff
@@ -41,6 +42,16 @@ export const MetricsExplorerPage = injectI18n(
       handleLoadMore,
     } = useMetricsExplorerState(source, derivedIndexPattern);
 
+    // Generate an index pattern if one doesn't exist the moment the app starts.
+    // This will prevent clicking on "Open in visualize" from redirecting to the
+    // index pattern page
+    useEffect(
+      () => {
+        fetch.get('../api/infra/generate_index_pattern');
```

Comment:
Also, the more I think about this, the more I wonder if we should pass the index pattern name in so that we make it easier to make one for logs later if necessary (possibly others too?). The problem is then to make it configurable we have to read from config in the client ... but that's not the worst problem. Thoughts? Alternatively we could make separate endpoints for each but that feels like more overhead...

Reply:
With the concerns that were brought up in the comments on this PR, I think we don't need to worry too much about making this easily extensible, so we can leave it hard-coded in the server for now.